interview-questions.md

```diff
@@ -8,44 +8,96 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+Researched answer: A relational database is stores and provides access points that are related to one another. Each row in a relational database table has a unique ID called the key. Columns in a relational database hold attributes of the data, with each record usually having a value for each attribute. Relational modeled databases are the best at maintaining data continuity across applications and instances of the same database. A good example of a relational db is a shopping cart when shopping online. when adding items, data that exists outside of the cart can be added to one users unique instance of the data. Other types of databases include: (just a brief description of each here, didnt want to over do it)
+
+        - Centralized database
+            - operates entirely in one location, users can access the database through a network connected to that db
+        - Cloud database
+            - data is stored on a local hd or server, users can access the data from anywhere provided they have an internet connection. Encryption is a major focus
+        - Commercial database
+            - designed by a commercial business, typically include a lot of useful features so it can be sold to customers.
+        - Distributed database
+            - db that is spread over multiple devices, helps increase the speed of the db, reliability and ease of expansion
+        - End-user database
+            - db that is primarily used my a single user
+        - Graph database
+            - focuses on data and connections between data. connections are stored with the data itself, providing a more efficient and faster db.
+        - NoSQL database
+            - data stored is not structured or relationa, allows for processing larger amounts of data, and makes the db easier to expand.
+        - Object-oriented database
+            - data is represented as objects and classes. this is a type of relational db. Each piece of data in this type of db is an object, and those objects grouped together are classes. useful for having large data that we want to process quickly
+        - Open-source database
+            - designed for public use for free. users can see how the program is written and are free to make changes to the program. typically lack more advanced features that are found in commercial databases.
+        - Operational database
+            - allows users to modify data in real time, can be relational or nosql. these dbs allow us to add, edit, or remove data anytime
+        - Personal database
+            - designed for a single user. not suitable for complex operations or businesses.
+
+(sources - https://www.oracle.com/database/what-is-a-relational-database/, https://www.indeed.com/career-advice/career-development/types-of-databases)
```

Comment:
Great job here, I definitely feel like you did a bit of research and you have an introduction to other databases that you may work with in the future. You may have seen or heard of MongoDB for example.